cpp/src/glm/qn_mg.cu

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "qn/glm_logistic.cuh"
+#include "qn/glm_regularizer.cuh"
+#include "qn/qn_util.cuh"
+#include "qn/simple_mat/dense.hpp"
+#include <cuml/common/logger.hpp>
+#include <cuml/linear_model/qn.h>
+#include <cuml/linear_model/qn_mg.hpp>
+#include <raft/core/comms.hpp>
+#include <raft/core/handle.hpp>
+#include <raft/util/cudart_utils.hpp>
+using namespace MLCommon;
+
+#include "qn/glm_base_mg.cuh"
+
+#include <cuda_runtime.h>
+#include <iostream>
+
+namespace ML {
+namespace GLM {
+namespace opg {
+
+template <typename T>
+void qnFit_impl(const raft::handle_t& handle,
+                const qn_params& pams,
+                T* X,
+                bool X_col_major,
+                T* y,
+                size_t N,
+                size_t D,
+                size_t C,
+                T* w0,
+                T* f,
+                int* num_iters,
+                size_t n_samples,
+                int rank,
+                int n_ranks)
+{
+  switch (pams.loss) {
+    case QN_LOSS_LOGISTIC: {
+      ASSERT(C == 2, "qn_mg.cu: logistic loss invalid C");
```

Comment:
Since this solver would normally support more loss functions, can you add a quick note for the user here that only the LOGISTIC loss is supported currently? Also, please use `RAFT_EXPECTS` instead of `ASSERT`  so that we have consistent messaging.